pkg/engine/variables/vars.go

```diff
@@ -561,13 +564,20 @@ func replaceSubstituteVariables(document interface{}) interface{} {
 		return document
 	}
 
-	regex := regexp.MustCompile(`\{\{([^{}]*)\}\}`)
 	for {
-		if len(regex.FindAllStringSubmatch(string(rawDocument), -1)) > 0 {
-			rawDocument = regex.ReplaceAll(rawDocument, []byte(`placeholderValue`))
-		} else {
+		if len(regexElementIndex.FindAllStringSubmatch(string(rawDocument), -1)) == 0 {
```

Comment:
Did you consider using [`regexp.FindAllSubmatch`](https://pkg.go.dev/regexp#Regexp.FindAllSubmatch), and avoiding the conversion from `[]byte` to `string`?